Adds a SecretDict field that obfuscated values in a dictionary (#7885)
PrefectHQ/prefect: docs/concepts/blocks.md

# aws_access_key_id='AKIAJKLJKLJKLJKLJKLJK' aws_secret_access_key=SecretStr('**********') aws_session_token=None profile_name=None region_name=None
```

> For more complex secret values, you can also use the `SecretDict` field type provided by Prefect. This type will allow you to add a dictionary field to your block that will have values at all levels automatically obfuscated in the UI or in logs. This is useful for blocks where typing or structure of secret fields is not known until configuration time.

Reviewer: ```suggestion
There's  also use the `SecretDict` field type provided by Prefect. This type will allow you to add a dictionary field to your block that will have values at all levels automatically obfuscated in the UI or in logs. This is useful for blocks where typing or structure of secret fields is not known until configuration time.
```

Here's an example of a block that uses `SecretDict`:

```python
from typing import Dict

from prefect.blocks.core import Block
from prefect.blocks.fields import SecretDict

class SystemConfiguration(Block):
    system_secrets: SecretDict
    system_variables: Dict

system_configuration_block = SystemConfiguration(
    system_secrets={
        "password": "p@ssw0rd",
        "api_token": "token_123456789",
        "private_key": "<private key here>",
    },
    system_variables={
        "self_destruct_countdown_seconds": 60,
        "self_destruct_countdown_stop_time": 7,
    },
)
```

> `system_secrets` will be obfuscated if `system_configuration_block` is logged, but `system_variables` will be shown in plain-text:

Reviewer 2: ```suggestion
`system_secrets` will be obfuscated when `system_configuration_block` is displayed, but `system_variables` will be shown in plain-text:
```

```python
print(system_configuration_block)

> # SystemConfiguration(system_secrets=SecretDict('{'password': '**********', 'api_token': '**********', 'private_key': '**********'}'), system_variables={'self_destruct_countdown_seconds': 60, 'self_destruct_countdown_stop_time': 7})

Reviewer 2: fwiw I'd use `pprint`

Author: I tried using `pprint` just now and it's still not pretty

Reviewer 2: Oh well, I often will add manual line breaks and indentation so it's readable in the doc even if it's not the real output
